networking/cluster_network_overview.py

```diff
@@ -0,0 +1,312 @@
+        availability_zone = subnet_info['AvailabilityZone']
+        cidr_block = subnet_info['CidrBlock']
+        total_ips = 2 ** (32 - int(subnet_info['CidrBlock'].split('/')[1])) - 5
+        available_ips = subnet_info['AvailableIpAddressCount']
+        used_ips = total_ips - available_ips
+        percentage_used = (used_ips / total_ips) * 100
+        percentage_free = 100 - percentage_used
+
+        return [
+            availability_zone,
+            cidr_block,
+            total_ips,
+            used_ips,
+            available_ips,
+            create_usage_bar(used_ips, total_ips)
+        ]
+    except Exception as e:
+        print(f"Error fetching subnet usage info for subnet {subnet_id}: {str(e)}")
+        return ["N/A"] * 6
+
+def get_cluster_subnet_data(eks_client, ec2_client, cluster_name):
+    print(f"Fetching subnet data for cluster {cluster_name}...")
+    try:
+        cluster_info = eks_client.describe_cluster(name=cluster_name)
+        cluster_subnet_ids = cluster_info['cluster']['resourcesVpcConfig']['subnetIds']
+
+        cluster_subnet_info = defaultdict(list)
+        for subnet_id in cluster_subnet_ids:
+            cluster_subnet_info[subnet_id] = get_subnet_usage_info(ec2_client, subnet_id)
+
+        return cluster_subnet_info
+    except Exception as e:
+        print(f"Error fetching cluster subnet data for cluster {cluster_name}: {str(e)}")
+        return defaultdict(list)
+
+
+def get_cluster_worker_node_ips(v1, ec2_client):
+    print("Fetching EC2 IPs for the EKS cluster...")
+    worker_node_ips = set()
+    subnet_ids = set()
+    worker_subnet_info = defaultdict(list)
+
+    try:
+        # List all nodes in the cluster
+        nodes = v1.list_node(watch=False)
+
+        # Iterate through nodes and get their private IP addresses, instance id, subnet id
+        for node in nodes.items:
+            for address in node.status.addresses:
+                if address.type == 'InternalIP':
+                    worker_node_ips.add(address.address)
+                    try:
+                        subnet_id = get_worker_node_subnets(ec2_client, address.address)
+                        subnet_ids.add(subnet_id)
+                    except Exception as e:
+                        print(f"Error getting subnet for worker node {address.address}: {str(e)}")
+
+        for subnet_id in subnet_ids:
+            try:
+                worker_subnet_info[subnet_id] = get_subnet_usage_info(ec2_client, subnet_id)
+            except Exception as e:
+                print(f"Error getting subnet usage info for subnet {subnet_id}: {str(e)}")
+                worker_subnet_info[subnet_id] = ["N/A"] * 6
+
+    except Exception as e:
+        print(f"Error fetching cluster worker node IPs: {str(e)}")
+
+    return worker_node_ips, worker_subnet_info
+
+def get_cluster_pod_ips(v1):
+    print("Fetching Pod IPs...")
+    pod_ips = set()
+
+    try:
```

Comment:
added loop to fetch all pods for using " _continue" and a "continue_token" since the call "list_pod_for_all_namespaces" is paginated. Using "metadata._continue" to check if there is more data to retrieve.